### PR TITLE
sqitch: 1.1.0 -> 1.3.1

### PR DESCRIPTION
**Pull request by johnrichardrinehart**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -915,11 +915,11 @@ let
   };
 
   AppSqitch = buildPerlModule {
-    version = "1.1.0";
+    version = "1.3.1";
     pname = "App-Sqitch";
     src = fetchurl {
-      url = "mirror://cpan/authors/id/D/DW/DWHEELER/App-Sqitch-v1.1.0.tar.gz";
-      hash = "sha256-7hRs111jAIN+bKVZuwveJH1CEjyWssXUsoAPONPj0as=";
+      url = "mirror://cpan/authors/id/D/DW/DWHEELER/App-Sqitch-v1.3.1.tar.gz";
```
Comment (lilyinstarlight) [inline pkgs/top-level/perl-packages.nix:921]: Minor comment, but it seems common for Perl packages to not interpolate `${version}` in the URLs and I don't really know why. A handful do it though so maybe it's just a holdover from old-style

**Review by lilyinstarlight — APPROVED** [review]
I left one comment (but you seem to have used the same convention as most of the file), but this looks good either way

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
Comment (johnrichardrinehart) [reply pkgs/top-level/perl-packages.nix:921]: Yeah, I think that pattern in `perl-packages.nix` isn't so good. So, I'm trying to spread some improvement. But, if it's not recommended for some reason that I didn't consider then I'm happy to edit.

Comment (johnrichardrinehart) [reply pkgs/top-level/perl-packages.nix]: Hm, I actually wasn't doing what I said/thought I was. I pushed a commit to recycle `pname` and `version` based on this comment.

**Comment by johnrichardrinehart** [comment]
Closed in light of https://github.com/NixOS/nixpkgs/pull/211809

**Pull request closed without merging**